src/Components/Playlist/addToPlaylist.js

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useState } from "react";
+import { useDataContext } from "../../Context/data-context";
+import { videoExists } from "../Video/videoUtil";
+
+export const AddToPlaylist = ({ id }) => {
+  const {
+    state: { playlist },
+    dispatch,
+  } = useDataContext();
+
+  const [showModal, setShowModal] = useState(false);
+  const [listName, setListName] = useState("");
+
+  useEffect(() => {
+    const modal = document.querySelector(".modal-container");
+    window.onclick = event => {
+      if (event.target === modal) {
+          setShowModal(false);
+      }
+  }
+  },[showModal]);
+
+  const createPlaylistHandler = (event) => {
+    if (event.key === "Enter") {
+      dispatch({ type: "ADD_TO_NEW_PLAYLIST", payload: { id, listName } });
+      setListName("");
+    }
+  };
+
+  const checkInPlaylist = (id) =>
+    playlist.some((item) => item.videos.some((videoId) => videoId === id));
+
+  return (
+    <>
+      {checkInPlaylist(id) ? (
+        <i
+          onClick={() => setShowModal((showModal) => !showModal)}
+          className="btn fas fa-lg fa-check primaryBg-txt"
+        >
+          {" "}
+          Saved{" "}
+        </i>
+      ) : (
+        <i
+          onClick={() => setShowModal((showModal) => !showModal)}
+          className="btn fas fa-lg fa-plus"
+        >
+          {" "}
+          Save{" "}
+        </i>
+      )}
+      {showModal && (
+        <div className="modal-container">
+          <div className="card modal-content">
+            <h3 className="modal-header">Playlists</h3>
+            <hr />
+            <ul className="list-checkbox">
+              {playlist.map(({ listId, name, videos }) => (
+                <li key={name} className="secondary-txt">
+                  <label>
+                    <input
+                      type="checkbox"
+                      onChange={() =>
+                        dispatch({
+                          type: "TOGGLE_PLAYLIST",
+                          payload: { listId, id },
+                        })
+                      }
+                      checked={videoExists(videos, id)}
+                    />{" "}
+                    {name}
+                  </label>
+                </li>
+              ))}
+            </ul>
+            <div className="txt-box">
+              <input
+                onKeyDown={createPlaylistHandler}
```

Comment:
we should not user handler in function name, createPlaylist itself is explanatory